python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition_evaluator.py

```diff
@@ -23,68 +24,66 @@
 from dagster._core.definitions.declarative_automation.automation_condition import AutomationResult
 from dagster._core.definitions.declarative_automation.automation_context import AutomationContext
 from dagster._core.definitions.events import AssetKey
+from dagster._core.instance import DagsterInstance
+from dagster._time import get_current_datetime
 
 if TYPE_CHECKING:
-    import datetime
-
     from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
 
 
-@dataclass
 class AutomationConditionEvaluator:
     def __init__(
         self,
         *,
-        asset_graph: BaseAssetGraph,
         entity_keys: AbstractSet[EntityKey],
-        asset_graph_view: AssetGraphView,
-        logger: logging.Logger,
+        instance: DagsterInstance,
+        asset_graph: BaseAssetGraph,
         cursor: AssetDaemonCursor,
-        data_time_resolver: CachingDataTimeResolver,
-        respect_materialization_data_versions: bool,
-        # Mapping from run tags to values that should be automatically added run emitted by
-        # the declarative scheduling system. This ends up getting sources from places such
-        # as https://docs.dagster.io/deployment/dagster-instance#auto-materialize
-        # Should this be a supported feature in DS?
-        auto_materialize_run_tags: Mapping[str, str],
-        request_backfills: bool,
+        evaluation_time: Optional[datetime.datetime] = None,
+        logger: logging.Logger = logging.getLogger("dagster.automation"),
     ):
-        self.asset_graph = asset_graph
         self.entity_keys = entity_keys
-        self.asset_graph_view = asset_graph_view
+        self.asset_graph_view = AssetGraphView(
+            temporal_context=TemporalContext(
+                effective_dt=evaluation_time or get_current_datetime(),
+                last_event_id=instance.event_log_storage.get_maximum_record_id(),
```

Comment:
not new but storage fetch in constructor feels not ideal - some staticmethod that communicates that non trivial work is being done would be nice